dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunction.cs

```diff
@@ -352,6 +379,17 @@ await using (enumerator.ConfigureAwait(false))
         }
     }
 
+    /// <summary>
+    /// Creates a new <see cref="KernelFunction"/> object that is a copy of the current instance
+    /// but the <see cref="KernelFunctionMetadata"/> has the plugin name set.
+    /// </summary>
+    /// <param name="pluginName">The name of the plugin this function instance will be added to.</param>
+    /// <remarks>
+    /// This method should only be used to create a new instance of a <see cref="KernelFunction"/> when adding
+    /// a function to a <see cref="KernelPlugin"/>.
+    /// </remarks>
+    public abstract KernelFunction CreateForPlugin(string pluginName);
```

Comment:
I assume we don't support using version N of Microsoft.SemanticKernel.Abstractions but version N - 1 of Microsoft.SemanticKernel.Core. If we did, adding an abstract method to KernelFunction would be a breaking change, since even though the only things that can derive from it are things with internal access, a previous version of the Core assembly wouldn't have the necessary overrides.

I don't think this is a real problem, just calling it out for completeness.

Reply:
Yeah, we've decided to keep things simple in our versioning story and require that all versions of all packages are the same. This is the only combination we test and will officially support.